privacyidea/models.py

```diff
@@ -2327,6 +2327,27 @@ def increase(self):
         self.counter_value = self.counter_value + 1
         self.save()
 
+    def decrease(self, allow_negative=False):
+        """
+        Decrease the value of a counter, stop at zero if allow_negative not given
+        :param allow_negative:
+        :return:
+        """
+        if self.counter_value == 0 and not allow_negative:
+            # leave counter at zero, nothing to save
+            # TODO: what should be done if the counter is already negative?
```

Comment:
This would mean, that the event handler was changed after the counter decreased under zero by previous events.
So the current event handler definition would be: "Be not negative".

Maybe we should reset the counter in this case to zero?